src/Controller/EventController.php

```diff
@@ -66,4 +97,50 @@ public function showEvent(Event $event)
     {
         return ['event'=>$event];
     }
+
+    /**
+     * @Route("/event/edit/{id}/{token}", name="edit_event", requirements={"id"="\d+"})
+     * @Template("/admin/event/edit.html.twig")
+     */
+    public function editEvent(Request $request, Event $event, $token, FileUploader $fileUploader)
+    {
+        if ($event->getToken() != $token){
+            throw $this->createNotFoundException('Возможно, вы пытаетесь отредактировать ранее созданное событие, но это неправильная ссылка. 
+            В таком случае обратитесь к администратору.');
+        } else {
+
+            $initialPicture = $event->getPicture();
+            if ($initialPicture) {
+                $event->setPicture(
+                    new File($this->getParameter('pictures_directory') . '/' . $event->getPicture())
+                );
+            }
+            $form = $this->createForm(EventFormType::class, $event);
+            $form->handleRequest($request);
+
+            if ($form->isSubmitted() && $form->isValid()) {
+                $event = $form->getData();
+                $file = $form->get('picture')->getData();
+                if ($file) {
+                    $fileName = $fileUploader->upload($file);
+                    if ($fileName) {
+                        $event->setPicture($fileName);
+                    }
+                } else {
+                    $event->setPicture($initialPicture);
+                }
+
+                $this->saveEntity($event);
+
+                $this->addSuccessFlash(sprintf('Событие "%s" отредактировано.', $event->getName()));
+                $this->addNoticeFlash('Вы редактируете ранее созданное событие. <br>
```

Comment:
@daria-sieroshtan 
Two flashes in a row? That'w weird

Reply:
Indeed, as our life. 
I wanna them both.

Reply:
Indeed, as our life. 
I need both.

Reply:
So is our life. I need both.

Reply:
As is our life. I need both.